files/shaders/objects_fragment.glsl

```diff
@@ -47,7 +47,7 @@ varying float depth;
 #define PER_PIXEL_LIGHTING (@normalMap || @forcePPL)
 
 #if !PER_PIXEL_LIGHTING
-varying vec4 lighting;
+centroid varying vec4 lighting;
 #else
 varying vec4 passColor;
```

Comment:
Adding "centroid" to passColor in both shaders fixed the issue with per-pixel lighting too.